hipcheck/src/policy_exprs/env.rs

```diff
@@ -446,6 +510,30 @@ fn divz(env: &Env, args: &[Expr]) -> Result<Expr> {
 			Float(arg_1 / arg_2)
 		}),
 		(Bool(_), Bool(_)) => Err(Error::BadType(name)),
+		(DateTime(arg_1), DateTime(arg_2)) => Err(Error::BadType(name)),
+		(DateTime(arg_1), Span(arg_2)) => Err(Error::BadType(name)),
+		(Span(arg_1), Span(arg_2)) => Err(Error::BadType(name)),
+		_ => unreachable!(),
+	};
+
+	binary_primitive_op(name, env, args, op)
+}
+
+// Finds the difference in time between two datetimes, in units of hours (chosen for comparision safety)
+fn duration(env: &Env, args: &[Expr]) -> Result<Expr> {
+	let name = "duration";
+
+	let op = |arg_1, arg_2| match (arg_1, arg_2) {
+		(Int(_), Int(_)) => Err(Error::BadType(name)),
+		(Float(_), Float(_)) => Err(Error::BadType(name)),
+		(Bool(_), Bool(_)) => Err(Error::BadType(name)),
```

Comment:
In general, is it possible to replace a lot of these `(Type(_), Type(_)) => Error::BadType` with a single `(_, _) => Error::BadType`?

Reply:
I think that should work. Let me try it.

Reply:
Made the change. Let me know if that's what you were thinking.

Reply:
Yeah looks good, there are still a few `(Bool(_), Bool(_)) => Err(Error::BadType(name)),` floating around, such as in the `add` function impl

Reply:
Great, I'll do the rest!